cms/categories/models.py

```diff
@@ -26,7 +26,35 @@ class CategoryPageCategoryRelationship(models.Model):
 class CategoryPage(Page):
     """CategoryPages are pages which have categories, like Blogs and Publications."""
 
-    pass
+    def blurb(self, length=110):  # Length empirically chosen, feel free to change.
+        """
+        Create a short extract for displaying on the search page.
+        We would call it an excerpt or a snippet, but both those names are taken
+        (by a page field and a wagtail concept of smaller-than-page fragments)
+        """
+
+        body = self.specific.body
+        # Later we'll probably want to handle stream fields but this'll do for now.
+        assert (
+            type(body) == str
+        ), "Invalid assumption: body of page is a richtext string. "
+        # It's not *actually* html, so removing it with regular expressions is fine.
+        # We probably don't need to process the whole string, which could be quite long
+        # TODO: consider cutting at 1k or so and matching for <[^<]$ too.
+        no_tag_blurb = re.sub("\s*<[^<]+>\s*", " ", body).lstrip()
```

Comment:
It might be worth swapping this for [`django.template.defaultfilters.truncatechars_html`](https://github.com/django/django/blob/main/django/template/defaultfilters.py#L276) to avoid reimplementing the logic.

Reply:
That line would be replaced by django.utils.strip_tags, but that doesn't put spaces between tags (so `<h1>header</h1><p>body</p>` would render as `headerbody`.
Neither truncatechars nor truncatechars_html understand escapes. 
I'm not sure what we gain from these functions.

Reply:
Used truncatechars, dropped the space requirements, moved &...; escaping further up into function